src/modules/0.0.19/aws_ecs_simplified/sql/create_triggers.sql

```diff
@@ -0,0 +1,320 @@
+
+    SELECT security_group_id INTO _security_group_id
+    FROM service_security_groups
+    WHERE service_name = _service_name LIMIT 1;
+
+    is_valid = _app_name IS NOT NULL AND _app_port IS NOT NULL AND _security_group_id IS NOT NULL;
+
+    is_valid = is_valid AND 1 = (SELECT COUNT(*) FROM security_group_rule WHERE security_group_id = _security_group_id AND is_egress = false AND ip_protocol = 'tcp' AND from_port = _app_port AND to_port = _app_port AND cidr_ipv4 = '0.0.0.0/0');
+
+    is_valid = is_valid AND 1 = (SELECT COUNT(*) FROM security_group_rule WHERE security_group_id = _security_group_id AND is_egress = true AND ip_protocol = '-1' AND from_port = -1 AND to_port = -1 AND cidr_ipv4 = '0.0.0.0/0');
+
+    SELECT cpu_memory::TEXT
+    INTO _cpu_mem
+    FROM task_definition
+    WHERE id = _task_definition_id AND execution_role_name = task_role_name;
+
+    -- TODO get latest or sort ?
+    SELECT tag, digest, repository_name, log_group_name, cpu, memory, image, env_variables
+    INTO _image_tag, _image_digest, _repository_name, _log_group_name, _cpu, _mem, _repository_uri, _env_variables
+    FROM container_definition
+    WHERE task_definition_id = _task_definition_id AND host_port = _app_port AND container_port = _app_port AND essential = true LIMIT 1;
+
+    IF _cpu_mem IS NULL THEN
+      _cpu_mem = get_cpu_mem_enum_from_parts(_cpu, _mem);
+    END IF;
+
+    IF _repository_name IS NOT NULL THEN
+      SELECT repository_uri INTO _repository_uri FROM repository WHERE repository_name = _repository_name;
+    END IF;
+
+    is_valid = is_valid AND 1 = (SELECT COUNT(*) FROM listener WHERE port = _app_port AND target_group_name = _target_group_name AND protocol = 'HTTP' AND action_type = 'forward');
+
+    SELECT load_balancer_name INTO _load_balancer_name
+    FROM listener
+    WHERE port = _app_port AND target_group_name = _target_group_name;
+
+    is_valid = is_valid AND 1 = (SELECT COUNT(*) FROM load_balancer WHERE load_balancer_name = _load_balancer_name AND scheme = 'internet-facing' AND load_balancer_type = 'application' AND ip_address_type = 'ipv4');
+
+    is_valid = is_valid AND 1 = (SELECT COUNT(*) FROM load_balancer_security_groups WHERE load_balancer_name = _load_balancer_name AND security_group_id = _security_group_id);
+
+    SELECT dns_name INTO _load_balancer_dns
+    FROM load_balancer
+    WHERE load_balancer_name = _load_balancer_name;
+
+    IF is_valid THEN
+      INSERT INTO ecs_simplified (app_name, desired_count, app_port, cpu_mem, image_tag, public_ip, load_balancer_dns, repository_uri, env_variables, force_new_deployment)
+      VALUES (_app_name, _desired_count, _app_port, _cpu_mem::ecs_simplified_cpu_mem_enum, _image_tag, _public_ip, _load_balancer_dns, _repository_uri, _env_variables, _force_new_deployment);
+    END IF;
+  END LOOP;
+
+  RETURN NULL;
+END
+$$;
+
+CREATE TRIGGER ecs_simplified_service_trigger
+AFTER INSERT OR DELETE OR UPDATE ON service
```

Comment:
my last question would be, what happens when any other table is edited that might cause a misconfiguration?

Reply:
good point :) I realized that today and I was going to address that in a follow up PR since it requires adding more tests to check for these changes and I was afraid it would be too hard to follow, plus it had already dragged on for a while

Reply:
I'm good with that, just something to be aware

Reply:
Actually, I just added it in this PR